File: src/main/java/no/nav/veilarboppfolging/config/DatabaseMigrator.java
package no.nav.veilarboppfolging.config;

import com.zaxxer.hikari.HikariConfig;
import com.zaxxer.hikari.HikariDataSource;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.flywaydb.core.Flyway;
import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;

import jakarta.annotation.PostConstruct;
import javax.sql.DataSource;

@Slf4j
@Component
@RequiredArgsConstructor
public class DatabaseMigrator {

    private final DataSource flywayDataSource;

    @PostConstruct
    public void migrateDb() {
        log.info("Starting database migration...");
        var flyway = new Flyway(Flyway.configure()
                .dataSource(flywayDataSource)
                .table("schema_version"));
        flyway.migrate();
    }

}

File: src/main/kotlin/no/nav/veilarboppfolging/dbutil/DatabaseMigrator.kt
package no.nav.veilarboppfolging.dbutil

import org.flywaydb.core.Flyway
import org.slf4j.LoggerFactory
import javax.sql.DataSource

val log = LoggerFactory.getLogger("DatabaseMigrator")
fun migrateDb(flywayDataSource: DataSource) {
    log.info("Starting database migration...")
    val flyway = Flyway(
        Flyway.configure()
            .dataSource(flywayDataSource)
            .table("schema_version")
    )
    flyway.migrate()
}